src/pip/_internal/network/auth.py

```diff
@@ -0,0 +1,289 @@
+import logging
```

Comment:
Let's add a doc-string to the module describing what all is happening in it and what we expect would be used by "consumers" of the module.

In general, when we're adding new modules as a result of breaking up existing modules, we should do this so that it makes our lives easier 6 months later when we forget how exactly we'd split the functionality. :)

Reply:
Sure. I'll create a follow-up PR later today.

Reply:
Filed #7054 for you. :)